django/middleware/csrf.py

```diff
@@ -160,23 +161,52 @@ def _reject(self, request, reason):
         )
         return _get_failure_view()(request, reason=reason)
 
-    def process_view(self, request, callback, callback_args, callback_kwargs):
-        if getattr(request, 'csrf_processing_done', False):
-            return None
-
+    def _get_current_token(self, request):
         try:
-            cookie_token = request.COOKIES[settings.CSRF_COOKIE_NAME]
+            if settings.CSRF_USE_SESSIONS:
+                if not getattr(request, 'session', None):
```

Comment:
Maybe change that into a `try/execpt AttributeError`. It's kinda nitpicky, but given that if you want to use session-based CSRF you will most likely have a session object on the request and then try/except would be faster (And even if not, it seems more natural and shows a nice chained error on python3).